### PR TITLE
July 2024 Deepgram product release update

### PR DESCRIPTION
**Pull request by bd-g**
## Proposed changes

- Update Helm chart and other resources to `release-240725` of Deepgram containers
  - Expose feature flags for three new self-hosted beta features: pre-recorded entity detection, improved pre-recorded  redaction, and improved streaming formatting

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - Removed certain Helm options for feature flags that should always be enabled 
- [ ] Documentation update or tests (if none of the other choices apply)

## Checklist

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] I have tested my changes in my local self-hosted environment
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: charts/deepgram-self-hosted/CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,20 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.1.0/),
 
 ## [Unreleased]
 
+## [0.4.0] - 2024-07-25
+
+### Added
+- Introduced entity detection feature flag for API containers (`false` by default).
+- Updated default container tags to July 2024 release. Refer to the [main Deepgram changelog](https://deepgram.com/changelog/deepgram-self-hosted-july-2024-release-240725) for additional details. Highlights include:
+  - Support for Deepgram's new English/Spanish multilingual code-switching model
+  - Beta support for entity detection
```
Comment (jkroll-deepgram) [inline charts/deepgram-self-hosted/CHANGELOG.md:15]: Specify "for pre-recorded audio" again here to be consistent with the other bullets in this list?

File: charts/deepgram-self-hosted/CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,20 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.1.0/),
...
+  - Beta support for improved redaction for pre-recorded audio
+  - Beta support for improved entity formatting for streaming audio
```
Comment (jkroll-deepgram) [inline charts/deepgram-self-hosted/CHANGELOG.md:17]: Worth specifying that these are all English-only?

File: charts/deepgram-self-hosted/CHANGELOG.md (hunk 1 above)
Comment (bd-g) [reply charts/deepgram-self-hosted/CHANGELOG.md]: 2873794d5e84

File: charts/deepgram-self-hosted/CHANGELOG.md (hunk 2 above)
Comment (bd-g) [reply charts/deepgram-self-hosted/CHANGELOG.md]: 2873794d5e84

**Review by jkroll-deepgram — APPROVED** [review]

**Pull request merged**